lib/internal/http2/compat.js

```diff
@@ -97,43 +97,45 @@ function onStreamError(error) {
 }
 
 function onRequestPause() {
-  this[kStream].pause();
+  const stream = this[kStream];
+  if (stream)
+    stream.pause();
 }
 
 function onRequestResume() {
-  this[kStream].resume();
-}
-
-function onRequestDrain() {
-  if (this.isPaused())
-    this.resume();
+  const stream = this[kStream];
+  if (stream)
+    stream.resume();
 }
 
-function onStreamResponseDrain() {
+function onStreamDrain() {
   this[kResponse].emit('drain');
 }
 
+// TODO Http2Stream does not emit 'close'
 function onStreamClosedRequest() {
   this[kRequest].push(null);
 }
 
+// TODO Http2Stream does not emit 'close'
 function onStreamClosedResponse() {
-  const res = this[kResponse];
-  res.writable = false;
-  res.emit('finish');
+  const response = this[kResponse];
+  response.writable = false;
+  response.emit('finish');
 }
 
 function onStreamAbortedRequest(hadError, code) {
-  if ((this.writable) ||
-      (this._readableState && !this._readableState.ended)) {
-    this.emit('aborted', hadError, code);
-    this.emit('close');
+  const request = this[kRequest];
+  if (request[kState].closed === false) {
+    request.emit('aborted', hadError, code);
+    request.emit('close');
   }
 }
 
 function onStreamAbortedResponse() {
-  if (this.writable) {
-    this.emit('close');
+  const response = this[kResponse];
+  if (response[kState].closed === false) {
```

Comment:
`writable` is currently always true since `Http2Stream` doesn't emit a 'close' event but also this seems more correct, as far as I can reason about it and compare it to http1.

Reply:
👍